### ISSUE TITLE
MapInfo widget _setScale error

### ISSUE THREAD

**Issue by roemhildtg** [cmv/cmv-app#381]
An error occurs occasionally when the map loads in the MapInfo widget.
`TypeError: Cannot read property 'decimalFormat' of undefined`
 It doesn't happen regularly, but about every other time... I've tracked it down to dojo.number. To replicate set the mapInfo options to something like this:

```JavaScript
                    proj4Catalog: 'EPSG', //'ESRI', 'EPSG' or 'SR-ORG' **
                    proj4Wkid: 3435, //wkid of the map ** // 4326
                    mode: 'dec', //'map', 'dec' or 'dms' // map
                    ...
```

Maybe I'm just doing something wrong..here's my entire widget config. Also, the basemap is set to web mercator.
```Javascript 
            mapInfo: {
                include: true,
                id: 'mapInfo',
                type: 'domNode',
                path: 'gis/dijit/MapInfo',
                srcNodeRef: 'mapInfoDijit',
                options: {
                    map: true, //required
                    mode: 'dec', //'map', 'dec' or 'dms' // map
                    firstCoord: 'y', //which coord to display first ('x')
                    unitScale: 2, //coord decimal places (2)(affects seconds in 'dms' format)
                    showScale: true, //show map scale (false)
                    showZoom: false, //show zoom level (false)
                    xLabel: 'X: ', //label for x coord ('X:')
                    yLabel: 'Y: ', //label for y coord ('Y:')
                    scaleLabel: '1:', //label for map scale ('1:')
                    zoomLabel: 'Z', //label for zoom level ('Z')
                    minWidth: 286, //minimum width in pixels of widget (0)(when 0 widget fits content)
                    proj4Catalog: 'EPSG', //'ESRI', 'EPSG' or 'SR-ORG' **
                    proj4Wkid: 3435 //wkid of the map ** // 4326
                }
            },
```

**Comment by darinatch** [comment]
@roemhildtg, 

I have experienced the same issue, and I think it comes down to the application not being able to connect to the spatialreference.org. It could be your connection or the site. 
And @tmcgee just posted in the chat a link how to host a projection on your server [here] ( http://docs.cmv.io/en/1.3.3/widgets/StreetView/). 
I will try it and see if I will still get this error.

**Comment by darinatch** [comment]
@roemhildtg, 

You are right, the problem comes from dojo's number.format. If removed and the scale number is only rounded, the issue does not persist.

**Comment by faridcher** [comment]
i have this problem too. when this bug will be solved?

**Comment by btfou** [comment]
I'm not sure what the issue with number.format is.

@roemhildtg @darinatch @faridcher What do think about just removing it? Small scales (large numbers) might be hard to read.

**Comment by faridcher** [comment]
@btfou, I really need it during development and extracting extents and other stuffs.

**Comment by roemhildtg** [comment]
@faridcher I think @btfou wasn't referring to the widget but the dojo.number call inside the widget since it is throwing errors. 

I support this since it solves the issue but it would be nice to know why the error occurs.  Last time I looked into it it seemed like it was trying to format a null,  but when I tried logging the value, the error was not reproducible.

**Comment by tmcgee** [comment]
A number of months ago, I removed the call to `number.format` for all the apps at a client site and the problem went away.

I believe that the error occurs when trying to format a null value returned by this.map.getScale() which seems to happen on occasion when the map first loads. (timing issue?)

If we check for a null value first or use NaN to ensure we have a number, I am pretty confident the issue would be resolved.

**Comment by tmcgee** [comment]
something like this:
``` javascript
_setScale: function () {
    var scale = this.map.getScale();
    if isNaN(scale) {
        return;
    }
    html.set(this.scaleNode, String(number.format(number.round(scale, 0))));
},
```

**Comment by tmcgee** [comment]
fixed in #433
